### PR TITLE
Create HPOS tables if they do not exist when enabling compatibility mode

### LINKED ISSUE #51940: wp wc hpos compatibility-mode enable -> Error: HPOS tables do not exist.
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Run WP CLI `wp wc hpos compatibility-mode enable`
can result in 
>Error: HPOS tables do not exist.

if the HPOS tables aren't created yet.

It should create the HPOS tables however (or at least provide a flag to create them)

### Expected behavior

No error

### Actual behavior

>Error: HPOS tables do not exist.

### Steps to reproduce

Run WP CLI `wp wc hpos compatibility-mode enable`

### WordPress Environment

WooCommerce 9.3

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by Mayisha**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

If HPOS tables do not exist yet and we run the WP CLI command `wp wc hpos compatibility-mode enable`, it gives the following error-

> Error: HPOS tables do not exist.

- Creating the HPOS tables first, before enabling the compact mode if tables do not exist when `enable` command is run.
- No changes are made for the `disable` command.

Closes #51940 

### How to test the changes in this Pull Request:
- Delete the HPOS tables from your DB.
- Run the WP CLI command `wp wc hpos compatibility-mode enable` in `trunk` and you should see the error message.
- Run the WP CLI command `wp wc hpos compatibility-mode enable` in this branch.
    - You should not see any error message.
    - HPOS tables should be created in your DB.
    - Compatibility mode should be enabled.

### CODE REVIEW

File: plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php [hunk 1]
```diff
@@ -1350,7 +1350,11 @@ public function disable_compat_mode(): void {
 	 */
 	private function toggle_compat_mode( bool $enabled ): void {
 		if ( ! $this->synchronizer->check_orders_table_exists() ) {
-			WP_CLI::error( __( 'HPOS tables do not exist.', 'woocommerce' ) );
+			if ( $enabled ) {
+				$this->synchronizer->create_database_tables();
+			} else {
+				WP_CLI::error( __( 'HPOS tables do not exist. ' . $create_tables, 'woocommerce' ) );
```
Comment (albarin) [inline plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php]: Hey @Mayisha! I think this `$create_tables` variable is not defined.

Comment (Mayisha) [reply plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php]: Thanks @albarin 👍 
My bad! I missed to cleanup this part from an alternate approach I was trying. Fixed in [51e6e8b](https://github.com/woocommerce/woocommerce/pull/51973/commits/51e6e8bc5c2952e950e47950c31f7f90959011f0)

**Review by albarin — APPROVED** [review]
Thanks for the quick fix!

**Comment by Mayisha** [comment]
Hi @albarin 👋 

Thank you for the review. I am not authorized to merge this PR. Could you please help me with this?

**Pull request merged**